### DISCUSSION TITLE
Refs not unwrapped after reading from store

### DISCUSSION THREAD

**Discussion by szekelyisz** [vuejs/pinia#2708] [Help and Questions] [answered]
Consider the [this](https://play.pinia.vuejs.org/#eNqNVVtv2jAU/itWNAmQir2OqpNSqOimatoetmnsMS8hOVC38UW2Q5EY/33HdoCEwsZTknP7vnPhY5M8aE1XNSRpMraF4doRC67WpMrlcpIlzmbJfSa50Mo4siG1hYeiULV0M6cMkC1ZGCVIj7I8mqmzvUxmslDSYq0QNDlO6w98yKKWheNKesT+gGwySWIC3dVqnlhgI3MBKekJ6F0RmZLr7Z0P9yiqAlqpZb+TOkD3tgOyPICczepkk066gTcs3zW2HdSYxQnivPDDgdBV7gC/CBnPa+ewyrSoePGCc8VEHOwM3JhF1+kwJI1hX/4XFohg4C//bIWOWYtFcpXENQ5FrumzVRKXHtrJGgfuOo0Nehtehf/OkifntE0ZK0qJaSVUfGWoBMekFmyKYczgzLiAYanEdERv6IdbVnLr2nYKVgznRr1aMFglS65aOAyNKzBDA7IEA+ZS3KO0DvaR7zy+5pLnEZGF9/OhAbSElVOqssNc80uJtnOmt/SWXrOKzxkCMY781m8mMixBXFJ9Fzp9T69v6MdQdTVqigpf1RfF89zi+p3Fy1/w5dHyCyU0r8D80P7Su0eQV5V6/RZsztSwp1g8QfFywv5s15H1T3+TuIFWWy434Z69+3H2Hdb4vncKVdZVc3BnnHjdqqo9xxj2qZYl0m7FBbZfwy1zufxtH9cOpN015YmGaYT4MObP/2j9QHdER60pHoQOR7gXxhIWXEJXFMM19VAeUEClA7PICyCNDkacqGrWGaQbWKG0yVrMsauoKW/yZg5/zTG5IYKUD6+76n8IDidQKomXylAM1oFslOZjJZ+0W+j3moqoto3iIWxK+oM9RCQyuSf9ZmKn6LRIhPa2A3xE7mHgaZO7k1LMdU/cnvgD2Fe683uIy/DKm2z/AtdyUzk=) example. (Watch the browser console when using.)

The Set button inserts a value into the store, but when reading it back with the Get button, I'm getting the ref-wrapped version that it practically unusable later in the code as is. However, type checking suggests that the retrieved value is not ref-wrapped. I get an error if I try to unwrap it with `.value`. However, the console shows that it's a Proxy object rather than Account like code completion and the compiler thinks.

Is this expected behavior? Am I doing something wrong?

**Comment by posva** [comment, marked as answer]
It's a proxy object but still an object: you can use it just like that (without `.value`). This is just Vue. Try reading any value that is a primitive and you will get the primitive value instead of a proxy (because primitives are always copied in JS)

**Reply by szekelyisz** [reply]
Thanks, I realized that I confused Refs with Proxies.
